src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -145,7 +145,12 @@
      * on hold (for example while inspecting a suspicious state or to perform a maintenance
      * when a failure is predictable.)
      */
-    private /*almost final*/ AtomicBoolean paused = new AtomicBoolean();
+    private /*almost final*/ transient AtomicBoolean paused = new AtomicBoolean();
+
+    /**
+     * Persistent version of {@link #paused}.
+     */
+    private boolean executionPaused;
```

Comment:
Or maybe make it `Boolean` and only set it to a non-null value in `writeReplace` (and null it out in `readResolve`).

Reply:
No need for `volatile` since it is not read except during deserialization.

Considered setting it during `writeReplace` but seemed like overkill since it is only ever written in two places and it just as easy to set it then.